### ISSUE TITLE
Failing unit tests 50% InvalidCastException (CanCreatesNewJSONReader, CanReadJSONReader)

### ISSUE THREAD

**Issue by mechamecho** [mechamecho/DataDrivenTesting#6]
Trying to resolve the following exceptions

> Test Name: CanCreatesNewJSONReader()
> Test FullName:	TestDataAccess.Tests.JsonFile.CanCreatesNewJSONReader()
> Test Source:	C:\Users\Engineer\Desktop\Prep-eM\Data-Driven-testing\JsonReaderCsharp\DataDrivenTesting\TestDataAccess.Tests\UnitTest1.cs : line 31
> Test Outcome:	Failed
> Test Duration:	0:00:00.066
> Result Message:	
> Expected: No Exception to be thrown
>   But was:  <System.InvalidCastException: Unable to cast object of type 'Newtonsoft.Json.Linq.JValue' to type 'Newtonsoft.Json.Linq.JObject'.
>    at TestDataAccess.JSONReader.testData(JSONFile jsonFile, String testDataKey, Int32 index)
>    at TestDataAccess.JSONReader..ctor(JSONFile jsonFile, String testDataKey, Int32 testDataIndex)
>    at TestDataAccess.Tests.JsonFile.CreateJSONReader(JSONFile jsonFile, String testDataKey, Int32 testDataIndex)

 

> Test Name:  CanReadJSONReader()
> 
> Test Outcome:	Failed
> 
> Result Message:	System.InvalidCastException : Unable to cast object of type 'Newtonsoft.Json.Linq.JValue' to type 'Newtonsoft.Json.Linq.JObject'.
>

**Comment by dariusstrasel** [comment]
@mechamecho Im actually not getting any of these errors, as of 113b880551f067a9b887add23d46ab2f19e2e397

**Comment by dariusstrasel** [comment]
CanCreateNewJSONFile()
```
No Exception to be thrown
  But was:  <System.FormatException: File Path is not in the correct format or the File doesn't exist C:\Users\Engineer\Desktop\Prep-eM\Data-Driven-testing\JsonReaderCsharp\DataDrivenTesting\TestDataAccess.Tests\testData.json .
   at TestDataAccess.JSONFile..ctor(String fullPath)
   at TestDataAccess.Tests.JsonFile.CreateJSONFile(String filePath) in C:\Users\siruk\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\UnitTest1.cs:line 68
   at TestDataAccess.Tests.JsonFile.<>c.<CanCreateNewJSONFile>b__4_0() in C:\Users\siruk\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\UnitTest1.cs:line 26
   at NUnit.Framework.Constraints.ThrowsConstraint.VoidInvocationDescriptor.Invoke()
   at NUnit.Framework.Constraints.ThrowsConstraint.ExceptionInterceptor.Intercept(Object invocation)>
```

CanCreatesNewJSONReader()
```
Result StackTrace:	
at TestDataAccess.JSONFile..ctor(String fullPath)
   at TestDataAccess.Tests.JsonFile.CreateJSONFile(String filePath) in C:\Users\siruk\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\UnitTest1.cs:line 68
   at TestDataAccess.Tests.JsonFile.CanCreatesNewJSONReader() in C:\Users\siruk\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\UnitTest1.cs:line 33
Result Message:	System.FormatException : File Path is not in the correct format or the File doesn't exist C:\Users\Engineer\Desktop\Prep-eM\Data-Driven-testing\JsonReaderCsharp\DataDrivenTesting\TestDataAccess.Tests\testData.json .
```

CanReadJSONReader()
```
Result StackTrace:	
at TestDataAccess.JSONFile..ctor(String fullPath)
   at TestDataAccess.Tests.JsonFile.CanReadJSONReader() in C:\Users\siruk\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\UnitTest1.cs:line 46
Result Message:	System.FormatException : File Path is not in the correct format or the File doesn't exist C:\Users\Engineer\Desktop\Prep-eM\Data-Driven-testing\JsonReaderCsharp\DataDrivenTesting\TestDataAccess.Tests\testData.json .
```

**Comment by mechamecho** [comment]
Interesting, I am not getting these Exceptions, only the one that I posted earlier.

**Comment by mechamecho** [comment]
I know why you are getting it, I changed the fullPath string in the TestFixture to look like the path on my Computer. Sorry about that.

**Comment by dariusstrasel** [comment]
Yup! No worries. This has been a problem I haven't quite found a solution for in C#, but I researched and found some promising results. Ill push a PR with an update the constants in the Unit Test folder.

Did you notice the unit test solution uses a different version of .Net than the source project?

**Comment by dariusstrasel** [comment]
I am getting these same errors after PR#7 was merged.

**Comment by mechamecho** [comment]
I did, I don't know how to fix it though.

**Comment by mechamecho** [comment]
Issue resolved as of 27d231a14b093c50623a6b07d5f169f476097485
